libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -653,42 +645,55 @@ - (void)handleUserAgentResponse:(NSURL *)requestUrl {
 }
 
 - (NSString *)generateApprovalUrlString {
-    return [self generateApprovalUrlString:self.credentials];
+    return [self approvalURLForEndpoint:[self brandedAuthorizeURL]
+                            credentials:self.credentials
+                          webServerFlow:[[[SalesforceSDKManager sharedManager] appConfig] useWebServerAuthentication]
+                               protocol:nil
+                                 domain:nil];
 }
 
-- (NSString *)generateApprovalUrlString:(SFOAuthCredentials *)credentials {
-    NSAssert(nil != credentials.domain, @"credentials.domain is required");
+- (NSString *)approvalURLForEndpoint:(NSString *)authorizeEndpoint
```

Comment:
I tried to consolidate the approval url logic here, currently the intent is to keep the same url each path was generating before but there might be more consolidation possible?
- I wasn't sure why the IDP protocol was hardcoded to "https" vs credentials.protocol
- There was one case that didn't use the branded authorize endpoint but I wasn't sure if that was intentional